content/en/tracing/troubleshooting/go_compile_time.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Troubleshooting Go Compile-Time Instrumentation
+further_reading:
+- link: "/tracing/trace_collection/automatic_instrumentation/dd_libraries/go/"
+  tag: "Documentation"
+  text: "Tracing Go Applications"
+---
+
+## Overview
+
+This guide explains how to troubleshoot [Orchestrion][1]-managed builds. The procedures can help Datadog gather insights about build processes and can assist with bug reports.
+
+<div class="alert alert-warning">The generated files may contain sensitive project information such as source code and dependency names. If sharing such information publicly is a concern, contact Datadog support to share the data privately.</div>
+
+## Preserving the work tree
+
+Orchestrion records build transformations in the `go build` work tree. To prevent the `go` toolchain from cleaning this directory after building, use the `-work` flag:
+
+```shell
+orchestrion go build -work ./...
+WORK=/tmp/go-build2455442813
+```
+
+The work tree location prints at the start of the build, marked with `WORK=`. This directory contains subdirectories for each built `go` package.
+
+## Work tree contents
+
+When Orchestrion injects code into a source file, it writes the modified file to the package's stage directory in the `orchestrion/src` subdirectory. For modified package import configurations, the original file is preserved with a `.original` suffix. These human-readable files can be inspected to verify Orchestrion's actions. Contact Datadog support for help interpreting these files.
+
+## Logging configuration
+
+### Log levels
+
+Control Orchestrion's logging output using the `ORCHESTRION_LOG_LEVEL` environment variable or `--log-level` flag:
+
+| Level | Description |
+|-------|-------------|
+| `NONE`, `OFF` (default) | No logging output |
+| `ERROR` | Error information only |
+| `WARN` | Errors and warnings |
+| `INFO` | Errors, warnings, and informational messages |
+| `DEBUG` | Detailed logging |
+| `TRACE` | Extremely detailed logging |
+
+<div class="alert alert-warning">Setting <code>ORCHESTRION_LOG_LEVEL</code> to the <code>DEBUG</code> or <code>TRACE</code> levels may have a significant impact on build performance, and these settings are not recommended for normal operations.</div>
+
+### Log file output
+
+Direct logging messages to files instead of the console by setting the `ORCHESTRION_LOG_FILE` environment variable or `--log-file` flag with the desired file path.
```

Comment:
```suggestion
Write logging messages to files instead of the console by setting the `ORCHESTRION_LOG_FILE` environment variable or `--log-file` flag with the desired file path.
```